WUTokenHelper/main.cpp

```diff
@@ -17,20 +18,45 @@ extern "C" __declspec(dllexport) int  __stdcall GetWUToken(wchar_t** retToken) {
 	wprintf(L"Account count = %i\n", accounts.Size());
 	if (accounts.Size() == 0)
 		return WU_NO_ACCOUNT;
-	auto accountInfo = accounts.GetAt(0);
-	wprintf(L"ID = %s\n", accountInfo.Id().c_str());
-	wprintf(L"Name = %s\n", accountInfo.UserName().c_str());
 
+	hstring tokenBase64;
+
+	// Loop the accounts on the system incase
+	// the first doesn't have access to the token
+	for (size_t i = 0; i < accounts.Size(); i++)
+	{
+		auto accountInfo = accounts.GetAt(i);
+		wprintf(L"ID = %s\n", accountInfo.Id().c_str());
+		wprintf(L"Name = %s\n", accountInfo.UserName().c_str());
+
+		try {
+			TryGetToken(accountInfo, tokenBase64);
+		}
+		catch (...) {
```

Comment:
I don't like this. I'd be happier if it caught something specific.

Moreover, it shouldn't just silently swallow an error if one occurs.

Reply:
I'm no expert at C++ so wouldn't know the best way of doing that, could you implement it?